src/AppInstallerCLICore/Workflows/MultiQueryFlow.cpp

```diff
@@ -0,0 +1,153 @@
+        }
+
+        std::vector<std::unique_ptr<Execution::Context>> packageSubContexts;
+        auto& source = context.Get<Execution::Data::Source>();
+        for (const auto& query : *context.Args.GetArgs(Execution::Args::Type::MultiQuery))
+        {
+            AICLI_LOG(CLI, Info, << "Creating search query for package [" << query << "]");
+
+            // Search for the current package
+            SearchRequest searchRequest;
+
+            // Regardless of match type, always use an exact match for the system reference strings.
+            searchRequest.Inclusions.emplace_back(PackageMatchFilter(PackageMatchField::PackageFamilyName, MatchType::Exact, query));
+            searchRequest.Inclusions.emplace_back(PackageMatchFilter(PackageMatchField::ProductCode, MatchType::Exact, query));
+            searchRequest.Inclusions.emplace_back(PackageMatchFilter(PackageMatchField::Id, matchType, query));
+            searchRequest.Inclusions.emplace_back(PackageMatchFilter(PackageMatchField::Name, matchType, query));
+            searchRequest.Inclusions.emplace_back(PackageMatchFilter(PackageMatchField::Moniker, matchType, query));
+
+            auto searchContextPtr = context.CreateSubContext();
+            Execution::Context& searchContext = *searchContextPtr;
+            auto previousThreadGlobals = searchContext.SetForCurrentThread();
+
+            searchContext.Add<Execution::Data::Source>(source);
+            searchContext.Add<Execution::Data::SearchRequest>(std::move(searchRequest));
+
+            packageSubContexts.emplace_back(std::move(searchContextPtr));
+        }
+
+        context.Add<Execution::Data::PackageSubContexts>(std::move(packageSubContexts));
+    }
+
+    void SearchSubContextsForSingle::operator()(Execution::Context& context) const
+    {
+        std::vector<std::unique_ptr<Execution::Context>> packageSubContexts;
+        bool foundAll = true;
+
+        for (auto& searchContextPtr : context.Get<Execution::Data::PackageSubContexts>())
+        {
+            auto& searchContext = *searchContextPtr;
+            SearchRequest searchRequest = searchContext.Get<Execution::Data::SearchRequest>();
+            searchContext.Add<Execution::Data::SearchResult>(searchContext.Get<Execution::Data::Source>().Search(searchRequest));
+
+            switch (m_searchPurpose)
+            {
+            case SearchPurpose::Install:
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ false);
+                break;
+            case SearchPurpose::Upgrade:
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ true);
+                break;
+            case SearchPurpose::Uninstall:
+                searchContext <<
+                    Workflow::HandleSearchResultFailures <<
+                    Workflow::EnsureOneMatchFromSearchResult(true) <<
+                    Workflow::ReportPackageIdentity;

```

Comment:
Remove this one, looks like we are double reporting since it's reported in line 91 in UninstallMultiplePackages( and that is the better place to report since we started to do uninstall)